### PR TITLE
Feat: Allow Useragent to be passed to client via config

### PR DESCRIPTION
**Pull request by kassarl**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
> /kind flake

**What this PR does / why we need it**:
Allows UserAgent to be passed through cloud provider configuration, which will allow for customer attribution

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
UserAgents can now be passed as part of the cloud provider config
```

### CODE REVIEW

File: pkg/provider/azure.go [hunk 1]
```diff
@@ -781,6 +783,7 @@ func (az *Cloud) getAzureClientConfig(servicePrincipalToken *adal.ServicePrincip
 		Authorizer:              autorest.NewBearerAuthorizer(servicePrincipalToken),
 		Backoff:                 &retry.Backoff{Steps: 1},
 		DisableAzureStackCloud:  az.Config.DisableAzureStackCloud,
+		UserAgent:               az.Config.UserAgent,
```
Comment (nilo19) [inline pkg/provider/azure.go:786]: Could you update the doc as well if we need to update the configuration in azure.json?

Comment (feiskyer) [reply pkg/provider/azure.go:786]: this should be OK as the default value is still same as before

https://github.com/kubernetes-sigs/cloud-provider-azure/blob/0443bd214196ca00c06add9b9b31c5be63babf45/pkg/azureclients/armclient/azure_armclient.go#L88-L90

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**